cw-orch-daemon/src/senders/cosmos.rs

```diff
@@ -415,6 +417,25 @@ impl Wallet {
     }
 }
 
+// Helpers to facilitate some rare operations
+impl Wallet {
+    /// Uploads the `WasmPath` path specifier on chain.
+    /// The resulting code_id can be extracted from the Transaction result using [cw_orch_core::environment::IndexResponse::uploaded_code_id] and returns the resulting code_id
+    pub async fn upload_wasm(&self, wasm_path: WasmPath) -> Result<CosmTxResponse, DaemonError> {
+        let file_contents = std::fs::read(wasm_path.path())?;
+        let mut e = write::GzEncoder::new(Vec::new(), Compression::default());
+        e.write_all(&file_contents)?;
+        let wasm_byte_code = e.finish()?;
+        let store_msg = cosmrs::cosmwasm::MsgStoreCode {
+            sender: self.account_id(),
+            wasm_byte_code,
+            instantiate_permission: None,
+        };
+
+        self.commit_tx(vec![store_msg], None).await
+    }
+}
+
```

Comment:
Is it possible to move implementation from `DaemonAsyncBase::upload` here instead?

Reply:
So there will be left something like:
```rust
let wasm_path = <T as Uploadable>::wasm(self.chain_info());
self.sender.upload_wasm(wasm_path)
```

Reply:
Ok, added the `upload_wasm` function directly on the Sender trait, so it's not a CosmosSender specific thing anymore as well :)